optaplanner-core/src/main/java/org/optaplanner/core/config/util/ConfigUtils.java

```diff
@@ -347,14 +347,7 @@ public static int resolvePoolSize(String propertyName, String value, String... m
             Class<?> type, Type genericType,
             Class<? extends Annotation> annotationClass, String memberName) {
         if (!(genericType instanceof ParameterizedType)) {
-            throw new IllegalArgumentException("The " + parentClassConcept + " (" + parentClass + ") has a "
-                    + (annotationClass == null ? "auto discovered" : annotationClass.getSimpleName() + " annotated")
-                    + " member (" + memberName
-                    + ") with a member type (" + type
-                    + ") that returns a " + Collection.class.getSimpleName()
-                    + " which has no generic parameters.\n"
-                    + "Maybe the member (" + memberName + ") should return a typed "
-                    + Collection.class.getSimpleName() + ".");
+            return Object.class;
```

Comment:
The fail-fast is incorrect here. Raw `Collection` is not an untyped collection. It's `Collection<Object>`. This should not fail.

Reply:
Actually, I could use this argument for the entire method. It should never throw. It should just return `Object` - as the answer is always `Object`, unless we can detect a more specific answer.

Reply:
That's true unless that boolean autoDiscovered is true. Because with @PlanningSolution set to auto discover, IIRC, raw types don't work well.

Reply:
As discussed, we want to move the auto discover fail fast out of this method, into the SolutionDescriptor. This indeed should return Object.